Fecha: 09/02/2026

Nombre: Valeria Guzmán Galván

Expediente: 756902

# A03 - Factores cualitativos

## Parte I

Sin = 0

Con = 1

Grande = 1

Chica = 0

In [45]:
import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression
import matplotlib.pyplot as plt

In [46]:
from google.colab import files
uploaded = files.upload()

df = pd.read_excel("TablaAguasFrescas.xlsx")
df.head(9)

Saving TablaAguasFrescas.xlsx to TablaAguasFrescas (2).xlsx


Tamaño Hielo     Sabor  Score
0  Grande   Sin    Kahlúa    8.2
1  Grande   Con    Kahlúa    9.0
2   Chica   Sin    Kahlúa    7.0
3   Chica   Con    Kahlúa    7.5
4  Grande   Con  Horchata   10.0
5   Chica   Sin  Horchata    6.0
6   Chica   Con  Horchata    6.4
7  Grande   Sin  Horchata    9.5

In [47]:
df["Tamaño"] = df["Tamaño"].replace({
    "Chica": 0,
    "Grande": 1
})

/tmp/ipython-input-3137401903.py:1: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df["Tamaño"] = df["Tamaño"].replace({


In [48]:
df["Hielo"] = df["Hielo"].replace({
    "Sin": 0,
    "Con": 1
})

/tmp/ipython-input-962590589.py:1: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df["Hielo"] = df["Hielo"].replace({


In [49]:
df["Kahlúa"] = (df["Sabor"] == "Kahlúa").astype(int)
df["Horchata"] = (df["Sabor"] == "Horchata").astype(int)

In [50]:
df

Tamaño  Hielo     Sabor  Score  Kahlúa  Horchata
0       1      0    Kahlúa    8.2       1         0
1       1      1    Kahlúa    9.0       1         0
2       0      0    Kahlúa    7.0       1         0
3       0      1    Kahlúa    7.5       1         0
4       1      1  Horchata   10.0       0         1
5       0      0  Horchata    6.0       0         1
6       0      1  Horchata    6.4       0         1
7       1      0  Horchata    9.5       0         1

In [51]:
from scipy import stats
import statsmodels.api as sm

In [52]:
xf = df[["Tamaño","Hielo","Kahlúa","Horchata"]]
xf1 = sm.add_constant(xf)

y = df['Score'].values.reshape(-1,1)

In [53]:
xff = sm.add_constant(xf1)
model = sm.OLS(y,xff)
results = model.fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.837
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     6.824
Date:                Thu, 12 Feb 2026   Prob (F-statistic):             0.0473
Time:                        16:01:40   Log-Likelihood:                -6.6425
No. Observations:                   8   AIC:                             21.29
Df Residuals:                       4   BIC:                             21.60
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.3000      0.320     13.417      0.000       3.410       5.190
Tamaño         2.4500      0.555      4.414      0.012       0.909       3.991
Hielo          0.5500      0.555      0.991      0.378      -0.991       2.091
Kahlúa         2.1250      0.320      6.631      0.003       1.235       3.015
Horchata       2.1750      0.320      6.787      0.002       1.285       3.065
==============================================================================
Omnibus:                        7.172   Durbin-Watson:                   1.431
Prob(Omnibus):                  0.028   Jarque-Bera (JB):                1.242
Skew:                          -0.048   Prob(JB):                        0.537
Kurtosis:                       1.072   Cond. No.                     2.22e+16
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 3.37e-32. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

## Parte II:

Sin = -1

Con = 1

Grande = -1

Chica = 1

In [54]:
df_1 = df.copy()

In [55]:
df_1["Tamaño"] = df_1["Tamaño"].replace({
    0: 1,
    1: -1
    })

In [56]:
df_1["Hielo"]  = df_1["Hielo"].replace({
    0: -1,
    1: 1
    })

In [57]:
df_1

Tamaño  Hielo     Sabor  Score  Kahlúa  Horchata
0      -1     -1    Kahlúa    8.2       1         0
1      -1      1    Kahlúa    9.0       1         0
2       1     -1    Kahlúa    7.0       1         0
3       1      1    Kahlúa    7.5       1         0
4      -1      1  Horchata   10.0       0         1
5       1     -1  Horchata    6.0       0         1
6       1      1  Horchata    6.4       0         1
7      -1     -1  Horchata    9.5       0         1

In [61]:
xf_1 = df_1[["Tamaño","Hielo","Kahlúa","Horchata"]]
xf1_1 = sm.add_constant(xf_1)

y_1 = df_1['Score'].values.reshape(-1,1)

In [63]:
xff_1 = sm.add_constant(xf1_1)
model_1 = sm.OLS(y_1,xff_1)
results_1 = model_1.fit()
results_1.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.837
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     6.824
Date:                Thu, 12 Feb 2026   Prob (F-statistic):             0.0473
Time:                        16:04:49   Log-Likelihood:                -6.6425
No. Observations:                   8   AIC:                             21.29
Df Residuals:                       4   BIC:                             21.60
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          5.3000      0.185     28.644      0.000       4.786       5.814
Tamaño        -1.2250      0.278     -4.414      0.012      -1.996      -0.454
Hielo          0.2750      0.278      0.991      0.378      -0.496       1.046
Kahlúa         2.6250      0.293      8.973      0.001       1.813       3.437
Horchata       2.6750      0.293      9.143      0.001       1.863       3.487
==============================================================================
Omnibus:                        7.172   Durbin-Watson:                   1.431
Prob(Omnibus):                  0.028   Jarque-Bera (JB):                1.242
Skew:                          -0.048   Prob(JB):                        0.537
Kurtosis:                       1.072   Cond. No.                     1.84e+16
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 3.54e-32. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

## Parte III:

**Conclusión:**

Al comparar los resultados de la Parte I y la Parte II, se observa que cambiaron los valores del intercepto y algunos coeficientes. Esto ocurre porque al modificar los números con los que representamos las categorías, también cambia el punto de referencia del modelo.

Sin embargo, el valor de R2 se mantiene igual (0.837), al igual que el F-statistic y los demás indicadores generales. Esto significa que el modelo sigue explicando lo mismo y tiene la misma capacidad para ajustarse a los datos.

En conclusión, los cambios en los coeficientes se deben únicamente a la forma en que se codificaron las variables, pero el modelo en esencia es el mismo.